src/Core/Vipr.Core/CodeModel/OdcmProjection.cs

```diff
@@ -8,78 +8,237 @@
 
 namespace Vipr.Core.CodeModel
 {
-    public class OdcmProjection : IEquatable<OdcmProjection>
+    public class OdcmProjection
     {
+        public delegate OdcmCapability DefaultProvider(OdcmObject odcmObject, string termName);
+
+        private static IDictionary<string, DefaultProvider> _wellKnownCapabilities = new Dictionary<string, DefaultProvider>()
+        {
+            [TermNames.Update] = GetDefaultBooleanCapability,
+            [TermNames.UpdateLink] = GetDefaultNegativeLinkCapability,
+            [TermNames.Delete] = GetDefaultBooleanCapability,
+            [TermNames.DeleteLink] = GetDefaultNegativeLinkCapability,
+            [TermNames.Insert] = GetDefaultBooleanCapability,
+            [TermNames.Expand] = GetDefaultBooleanCapability,
```

Comment:
This code is mostly for backwards compatibility with Vipr.WriterCSharp.Lite; it might go away eventually.

I think we should try to keep Vipr as agnostic of any specific annotations as possible. It should be responsibility of OdcmWriter implementation to provide the correct defaults;
OdcmProjection.UserDefaultCapabilityProvider property can used for this purpose (see its usage in Given_a_valid_edmx_with_Capability_Annotations tests).

For now it's probably safe to assume that if BooleanValueOf() method returns null, the default is "true", but I can start working on changes to T4TemplateWriter to provide specific defaults.